File: 은지/LV1/체육복.py
def solution(n, lost, reserve):
    Cloth = [1] * (n+1)
    # 여벌의 체육복 입력
    for x in reserve:
        Cloth[x] += 1
    # 도난 체육복 입력
    for x in lost:
        Cloth[x] -= 1
    # 체육복 check!
    for i in range(1, n+1):
        # i번째 학생이 체육복이 없음
        if Cloth[i] < 1:
            if i-1 > 0 and Cloth[i-1] > 1:
                Cloth[i] += 1
                Cloth[i-1] -= 1
            elif i+1 <= n and Cloth[i+1] > 1:
                Cloth[i] += 1
                Cloth[i+1] -= 1
    # 결과값 초기화
    answer = -1
    for x in Cloth:
        if x > 0:
            answer += 1
    return answer

File: 은지/LV1/체육복.js
function solution(n, lost, reserve) {
    var answer = 0;
    var cloth = []
    for (let i=0; i<=n; i++) {
      cloth.push(1)
    }
    
    for (let i of lost) {
      cloth[i] -= 1
    }

    for (let i of reserve) {
      cloth[i] += 1
    }

    for (let i=0; i<=n; i++) {
      if (i-1>0 && cloth[i] === 2 && cloth[i-1] === 0) {
        cloth[i-1] = 1
        cloth[i] = 1
      }
      else if (i+1<=n && cloth[i] === 2 && cloth[i+1] === 0) {
        cloth[i+1] = 1
        cloth[i] = 1
      }
    }
    
    for (let i=1; i<n+1; i++) {
      if (cloth[i]) {
        answer += 1
      }
    }
    return answer
  }
n = 10
l = [8, 10]
r = [6, 7, 9]
console.log(solution(n, l, r))